src/client/app/component/HomePageHeader/index.js

```diff
@@ -35,21 +43,39 @@ class HomePageHeader extends React.Component {
   }
 
   render() {
-    const btnText = this.state.isLoggedIn ? 'Log Out' : 'Log In';
+    const userText = localStorage.getItem('Username') === null ?
+      'Log Out' : localStorage.getItem('Username');
+    const btnText = this.state.isLoggedIn ? userText : 'Log In';
+    let button = null;
+    const menu = (
+      <Menu onClick={this.onClickHeaderLogBtn.bind(this)}>
```

Comment:
can be done in the constructor
`this.onClickHeaderLogBtn = this.onClickHeaderLogBtn.bind(this);`